test/tests/contract.ts

```diff
@@ -75,275 +68,307 @@ const transferData = {
-                        data: newAccountData,
-                    },
-                    {
-                        name: 'buyrambytes',
-                        data: buyRamBytesData,
-                        authorization: [{actor: 'foo', permission: 'bar'}],
-                    },
-                    {
-                        name: 'delegatebw',
-                        data: delegateBwData,
-                    },
-                ])
-                assert.lengthOf(actions, 3)
-                assert.isTrue(actions[0].authorization[0].equals(PlaceholderAuth))
-                assert.lengthOf(actions[0].authorization, 1)
-                assert.isTrue(actions[1].authorization[0].actor.equals('foo'))
-                assert.isTrue(actions[1].authorization[0].permission.equals('bar'))
-                assert.lengthOf(actions[1].authorization, 1)
-                assert.isTrue(actions[2].authorization[0].equals(PlaceholderAuth))
-                assert.lengthOf(actions[2].authorization, 1)
+            test('throws for unknown action', async function () {
+                let error
+                try {
+                    await tokenContract.ricardian('foo')
+                } catch (err) {
+                    error = err
+                }
+                assert.instanceOf(error, Error)
             })
-        })
-    })
-
-    suite('actionNames', function () {
-        test('returns list', function () {
-            assert.isArray(systemContract.actionNames)
-            assert.lengthOf(systemContract.actionNames, 62)
-            assert.isTrue(systemContract.actionNames.includes('newaccount'))
-        })
-    })
-
-    suite('ricardian', () => {
-        test('returns ricardian contract', async function () {
-            const kit = new ContractKit({
-                client: makeClient('https://eos.greymass.com'),
+            test('throws for an undefined ricardian', async function () {
+                let error
+                try {
+                    await tokenContract.ricardian('transfer')
+                } catch (err) {
+                    error = err
+                }
+                assert.instanceOf(error, Error)
             })
-            const contract = await kit.load('eosio.token')
-            const ricardian = await contract.ricardian('transfer')
-            assert.isString(ricardian)
-        })
-        test('throws for unknown action', async function () {
-            let error
-            try {
-                await tokenContract.ricardian('foo')
-            } catch (err) {
-                error = err
-            }
-            assert.instanceOf(error, Error)
-        })
-        test('throws for an undefined ricardian', async function () {
-            let error
-            try {
-                await tokenContract.ricardian('transfer')
-            } catch (err) {
-                error = err
-            }
-            assert.instanceOf(error, Error)
         })
     })
-})
+})()
```

Comment:
This contract test file is getting pretty big, we should probably think about a way to break it up at some point.